packages/testing/src/chai-retry-plugin/chai-retry-plugin.ts

```diff
@@ -0,0 +1,122 @@
+             * - `timeout`: The maximum duration in milliseconds to wait before failing the retry operation.
+             * - `retries`: The number of times to retry the function before failing.
+             * - `delay`: The delay in milliseconds between retries.
+             * @default { timeout: 5000, delay: 0, retries: Infinity }
+             */
+            retry(options?: RetryOptions): PromiseLikeAssertion;
+        }
+    }
+}
+
+/**
+ * Adds the `retry` method to Chai assertions, which allows to check the return value of a function until it satisfies the chained assertions.
+ * Should be applied through `Chai.use` function, for example:
+ * @example
+ * ```ts
+ * import Chai from 'chai';
+ * import { chaiRetryPlugin } from '@wixc3/testing';
+ *
+ * Chai.use(chaiRetryPlugin);
+ * ```
+ *
+ * Examples of usage:
+ * @example
+ * ```ts
+ * await expect(funcToRetry).retry({ timeout: 7000, retries: 5 }).have.property('value').and.be.above(4);
+ * ```
+ * @example
+ * ```ts
+ * await expect(sometimesNullFunction).retry({ timeout: 2000 }).to.be.not.null;
+ * ```
+ * @example
+ * ```ts
+ * await expect(funcToRetry).retry({ retries: 5, delay: 10 }).and.have.property('success').and.be.true;
+ * ```
+ */
+export const chaiRetryPlugin = function (_: typeof Chai, utils: Chai.ChaiUtils) {
+    Assertion.addMethod('retry', function (retryOptions: RetryOptions = {}): PromiseLikeAssertion {
+        const functionToRetry: FunctionToRetry = this._obj as FunctionToRetry;
+
+        if (typeof functionToRetry !== 'function') {
+            throw new TypeError(utils.inspect(functionToRetry) + ' is not a function.');
+        }
+
+        const assertionStack: AssertionStackItem[] = [];
+        const defaultRetryOptions: Required<RetryOptions> = { timeout: 5000, retries: Infinity, delay: 0 };
+        const options: Required<RetryOptions> = { ...defaultRetryOptions, ...retryOptions };
+        /**
+         * This is needed to handle cases when function that user passes to `expect`
+         * should be called through the native chai's implementation rather then  within `chaiRetryPlugin`, for example:
+         *
+         * const myObj = { val: 1 },
+         *       addTwo = () => {
+         *              myObj.val += 2;
+         *          };
+         * await expect(addTwo).retry().to.increase(myObj, 'val').by(2);
+         */
+
+        // Fake assertion object for catching calls of chained methods
+        const proxyTarget = new Assertion({});
+
+        const assertionProxy: PromiseLikeAssertion = Object.assign(
+            new Proxy(proxyTarget, {
+                get: function (target: Chai.Assertion, key: string, proxySelf: Chai.Assertion) {
+                    let value: Chai.Assertion | undefined = undefined;
+                    try {
+                        value = target[key as keyof Chai.Assertion] as Chai.Assertion;
+                        // eslint-disable-next-line
+                    } catch (error) {
```

Comment:
no need (`error`) and `// eslint-disable-next-line` above

Reply:
also I think that line above `try` or above the code itself  is a better place for this comment. I mean you wan't to explain why there could even be an error in the `target[key]` and not why it's caught and not handled.

Reply:
just `catch {}`

Reply:
@yurii-ve fixed